From 45f68ce51f49e29ff7aa5046ea1b624355466157 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 14 Mar 2022 12:40:18 +0200
Subject: [PATCH] [SW-78698] gaudi3: update hashes for keys

Currently, the hash stored for gaudi3 are copied from gaudi2. This
patch updates hashes generated from gaudi3 pre-production and
production keys.

Change-Id: I983f15c55426dda9b3f8fb11cbaf733cdde2073d
---

diff --git a/arch/gaudi3/arc/security.c b/arch/gaudi3/arc/security.c
index 5318910..34bc845 100644
--- a/arch/gaudi3/arc/security.c
+++ b/arch/gaudi3/arc/security.c
@@ -11,9 +11,6 @@
 #include "spi_flash.h"
 #include <string.h>
 
-/* TODO: Add gaudi3 specific implementation */
-
-/* TODO: Update ROM keys for gaudi3*/
 /*
  * This key hash is calculated on the Public Key Properties -
  *   -----------------------------------------------
@@ -26,16 +23,16 @@
 const uint8_t rom_key_hashes[NUM_KEYS_IN_ROM][SHA384_SUM_LEN] = {
 	/* Habana Public Key */
 	{
-		0x08, 0x51, 0xe7, 0x07, 0xc1, 0x0a, 0xc8, 0x45,
-		0x51, 0x61, 0xb4, 0xdf, 0xb6, 0xd6, 0x89, 0xbc,
-		0x57, 0xbb, 0xf9, 0xe2, 0x50, 0x4f, 0x44, 0x1f,
-		0x12, 0x4a, 0x73, 0x3d, 0xbb, 0xda, 0x63, 0xf2,
-		0x76, 0x53, 0x6c, 0x12, 0xa1, 0x13, 0x37, 0x30,
-		0x03, 0xee, 0x79, 0xce, 0x20, 0xa3, 0x32, 0x33
+		0x95, 0xa3, 0x2a, 0x03, 0xd4, 0xbd, 0xb1, 0xe0,
+		0x2b, 0x54, 0xc0, 0x9c, 0x61, 0x10, 0x80, 0x2c,
+		0x08, 0xc4, 0x5d, 0xdf, 0x4a, 0xf8, 0x03, 0x00,
+		0x2a, 0xf5, 0xac, 0x20, 0x6f, 0x2c, 0x8f, 0x27,
+		0x69, 0x2e, 0x63, 0x7d, 0x02, 0x33, 0x54, 0x29,
+		0x3e, 0x57, 0x46, 0xd9, 0xf4, 0xa9, 0x7e, 0x88
 	},
-	/* Additional Keys*/
 	/* Production Key 1 */
 	{
+		/* TODO: Yet to receive production key pub key certificate */
 		0x47, 0x9b, 0x51, 0x5d, 0xf9, 0x8b, 0xfd, 0x3d,
 		0x38, 0x3f, 0x19, 0x51, 0x40, 0x62, 0xcb, 0x15,
 		0x2d, 0xae, 0x11, 0xe8, 0x1a, 0x91, 0x3e, 0x02,
@@ -45,30 +42,30 @@
 	},
 	/* Production Key 2 */
 	{
-		0x48, 0x3c, 0x50, 0xfc, 0xf1, 0x3a, 0x56, 0x7a,
-		0x7b, 0x7c, 0xcc, 0x77, 0x66, 0xb3, 0x85, 0x1b,
-		0x6f, 0x43, 0x60, 0xce, 0x16, 0x9a, 0xc4, 0x14,
-		0xd5, 0xcb, 0x11, 0x09, 0x66, 0xc0, 0xfe, 0x42,
-		0xfe, 0x78, 0x79, 0x57, 0xdb, 0x06, 0x5c, 0x37,
-		0x2f, 0xe2, 0x81, 0xb0, 0x00, 0x11, 0x9f, 0x8e,
+		0x0c, 0xbe, 0x41, 0xaa, 0x56, 0xd2, 0xe3, 0x29,
+		0x0f, 0x45, 0x2f, 0x48, 0x64, 0x1f, 0x61, 0x00,
+		0xa3, 0x94, 0x6c, 0x2b, 0xc0, 0xa5, 0xd0, 0x9a,
+		0x0d, 0x3d, 0xb3, 0x74, 0x85, 0x1c, 0xbc, 0x9b,
+		0xee, 0xc1, 0x3c, 0xbc, 0x7e, 0xae, 0xf0, 0x7c,
+		0x4a, 0x7e, 0x82, 0x52, 0x69, 0xbc, 0xdc, 0x5f
 	},
 	/* Production Key 3 */
 	{
-		0x5c, 0xc8, 0xd6, 0xf3, 0x42, 0xdc, 0xa3, 0xdf,
-		0xb0, 0xa3, 0xa8, 0x73, 0x68, 0x54, 0x78, 0x5e,
-		0x8c, 0xaf, 0x09, 0x0b, 0x0d, 0x41, 0xa8, 0xa1,
-		0xd9, 0xf8, 0x31, 0xdc, 0x95, 0x0f, 0x84, 0x02,
-		0xb3, 0x2a, 0x61, 0xe4, 0x10, 0xd8, 0x39, 0xdf,
-		0x80, 0x27, 0xc1, 0x2e, 0xd5, 0xd2, 0xb2, 0xcd,
+		0xd3, 0x21, 0x9d, 0xe5, 0xcf, 0x7e, 0x4c, 0x20,
+		0x98, 0x4e, 0x84, 0x4d, 0x00, 0x5f, 0xf8, 0x71,
+		0x98, 0xf6, 0x11, 0x8f, 0xd4, 0xfd, 0xf6, 0x29,
+		0x6a, 0xd3, 0xfd, 0x94, 0xcf, 0x21, 0xf5, 0x60,
+		0xe8, 0x78, 0x91, 0x4e, 0xc0, 0x51, 0xc6, 0x82,
+		0xe1, 0xae, 0xfe, 0xcf, 0xbf, 0x96, 0xcf, 0xaf
 	},
 	/* Production Key 4 */
 	{
-		0xdb, 0x8c, 0x8c, 0xe7, 0x08, 0x19, 0x6b, 0x04,
-		0x76, 0x43, 0xa7, 0xe4, 0x34, 0x2d, 0x2e, 0x8f,
-		0xf5, 0xdd, 0x20, 0xc9, 0xc7, 0x48, 0x0c, 0x4d,
-		0x80, 0x35, 0x18, 0x10, 0x8b, 0xb2, 0x04, 0xd2,
-		0x79, 0xc1, 0x6b, 0x33, 0x07, 0x99, 0x21, 0x3a,
-		0xbf, 0x77, 0xfa, 0x6b, 0x53, 0xf4, 0x27, 0x25,
+		0x98, 0xd0, 0x0c, 0xa3, 0xa4, 0xc3, 0xe8, 0x55,
+		0x8d, 0xcf, 0xd2, 0x48, 0x0d, 0x8a, 0x52, 0xd6,
+		0xf5, 0x4c, 0x45, 0x58, 0x6d, 0x20, 0xb9, 0xe2,
+		0x4f, 0xe7, 0x6d, 0x0b, 0xdc, 0x76, 0xf6, 0xad,
+		0x2f, 0x79, 0xca, 0x03, 0x92, 0xc9, 0x2a, 0x5f,
+		0x36, 0x19, 0xee, 0x61, 0x3a, 0x10, 0x69, 0xa9
 	}
 };
 
